subscriptions.md

```diff
@@ -0,0 +1,297 @@
+    "name" : "Frequency",
+    "value" : "Daily"
+  } ],
+  "subscriptionType" : "TestType",
+  "_links" : {
+    "dSpaceObject" : {
+      "href" : "http://localhost/api/core/subscriptions/1/dSpaceObject"
+    },
+    "ePerson" : {
+      "href" : "http://localhost/api/core/subscriptions/1/ePerson"
+    },
+    "self" : {
+      "href" : "http://localhost/api/core/subscriptions/1"
+    }
+  }
+}
+```
+
+Status codes:
+* 200 OK - if the subscription is found and it is visible to the current user.
+* 401 Unauthorized - if you are not authenticated and the subscription is not visible to anonymous users
+* 403 Forbidden - if you are not logged in with sufficient permissions. Only Admin and owner subscription has access.
+* 404 Not found - if the subscription doesn't exist
+
+### Search methods
+#### findByEPerson
+**GET /api/core/subscription/search/findByEPerson?uuid=<:ePerson-uuid>**
+
+The supported parameters are:
+* page, size [see pagination](README.md#Pagination)
+* uuid: mandatory, the uuid of the eperson that benefit of content subscription
+
+```json
+{
+  "_embedded": {
+    "subscriptions": [
+      {
+        "id": 60,
+        "type": "subscription",
+        "subscriptionParameterList" : [ {
+			    "id" : 2,
+			    "name" : "Frequency",
+			    "value" : "Daily"
+		  } ],
+        "subscriptionType": "subscription",
+        "_links": {
+          "dSpaceObject": {
+            "href": "http://localhost:8080/server/api/core/subscriptions/60/dSpaceObject"
+          },
+          "ePerson": {
+            "href": "http://localhost:8080/server/api/core/subscriptions/60/ePerson"
+          },
+          "self": {
+            "href": "http://localhost:8080/server/api/core/subscriptions/60"
+          }
+        }
+      }
+    ]
+  },
+  "_links": {
+    "self": {
+      "href": "http://localhost:8080/server/api/core/subscriptions/search/findByEPerson?id=49e5ff01-5467-4acd-be21-9ad7374be929"
+    }
+  },
+  "page": {
+    "size": 20,
+    "totalElements": 1,
+    "totalPages": 1,
+    "number": 0
+  }
+}
+```
+
+Status codes:
+* 200 OK - if the operation succeed
+* 400 Bad Request - if the uuid parameter is missing or invalid
+* 401 Unauthorized - if you are not authenticated
+* 403 Forbidden - if you are not logged in with sufficient permissions. Only system administrators or the user specified in the uuid parameter can use the endpoint
+
+
+#### findByEPersonAndDso
+**GET /api/core/subscriptions/search/findByEPersonAndDso?dspace_object_id={id}&eperson_id={id}>**
+Find all subscription that a person made for a dataSpaceObject
+
+The supported parameters are:
+* page, size [see pagination](README.md#Pagination)
+* eperson_id: mandatory, the uuid of the eperson that benefit of content subscription
+* dspace_object_id: mandatory, the uuid of the DSpaceObject (Community, Collection, Item)
+
+```json
+{
+  "subscriptions": [
+    {
+      "id": 60,
+      "type": "subscription",
+      "subscriptionParameterList" : [ {
+			    "id" : 12,
```

Comment:
it seems that the id is not really needed outside of the api, we cannot update a single parameter and it is conceptually not an entity maybe we can just avoid to expose over the rest api